### ISSUE TITLE
Database conflicts

### ISSUE THREAD

**Issue by slomo** [liqd/adhocracy3.mercator#1643]
Quite often I encounter database conflicts during deployment

```
ZODB.POSException.ConflictError: database conflict error (oid 0x01, class adhocracy_core.resources.pool.Pool, serial this txn started with 0x03b1ac382a6b6fcc, serial currently committed 0x03b1ac382a9a4955)
```

How can I prevent those: 
 * Should I not run the import scripts and the backend in parallel?
 * Should the evolve scripts not run while the backend does and vice versa?

**Comment by slomo** [comment]
I have another one. After deleting and evolving and importing again. It first works but then I restarted the server and a conflict occurred.

```
ZODB.POSException.ConflictError: database conflict error (oid 0x01, class adhocracy_core.resources.pool.Pool, serial this txn started with 0x03b1ac7a920f1788, serial currently committed 0x03b1ac7a92260266)
```

**Comment by slomo** [comment]
Here a trace:

```
Traceback (most recent call last):
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/arbiter.py", line 503, in spawn_worker
    worker.init_process()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/workers/base.py", line 116, in init_process
    self.wsgi = self.app.wsgi()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/app/wsgiapp.py", line 63, in load
    return self.load_pasteapp()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/app/wsgiapp.py", line 59, in load_pasteapp
    return load_pasteapp(self.cfgurl, self.relpath, global_conf=None)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/gunicorn/app/pasterapp.py", line 71, in load_pasteapp
    global_conf=global_conf)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/paste/deploy/loadwsgi.py", line 247, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/paste/deploy/loadwsgi.py", line 272, in loadobj
    return context.create()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/paste/deploy/loadwsgi.py", line 710, in create
    return self.object_type.invoke(self)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/paste/deploy/loadwsgi.py", line 146, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/adhocracy_kit/__init__.py", line 23, in main
    return config.make_wsgi_app()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/pyramid/config/__init__.py", line 951, in make_wsgi_app
    self.registry.notify(ApplicationCreated(app))
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/pyramid/registry.py", line 74, in notify
    [ _ for _ in self.subscribers(events, None) ]
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/zope/interface/registry.py", line 328, in subscribers
    return self.adapters.subscribers(objects, provided)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/zope/interface/adapter.py", line 600, in subscribers
    subscription(*objects)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/pyramid/config/adapters.py", line 103, in derived_subscriber
    return subscriber(arg[0])
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/adhocracy_kit/resources/subscriber.py", line 10, in set_root_acms
    set_acms_for_app_root(event.app, (kit_acm, root_acm))
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/adhocracy_core/authorization/__init__.py", line 164, in set_acms_for_app_root
    transaction.commit()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_manager.py", line 111, in commit
    return self.get().commit()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_transaction.py", line 280, in commit
    reraise(t, v, tb)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_compat.py", line 55, in reraise
    raise value
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_transaction.py", line 271, in commit
    self._commitResources()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_transaction.py", line 417, in _commitResources
  reraise(t, v, tb)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_compat.py", line 55, in reraise
    raise value
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/transaction/_transaction.py", line 394, in _commitResources
    rm.tpc_vote(self)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/ZODB/Connection.py", line 788, in tpc_vote
    s = vote(transaction)
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/ZEO/ClientStorage.py", line 1107, in tpc_vote
    return self._check_serials()
  File "/home/adhocracy3/adhocracy3/lib/python3.4/site-packages/ZEO/ClientStorage.py", line 937, in _check_serials
    raise s
```

Seems that all traces go through subscribers.py

**Comment by slomo** [comment]
I am trying to change the root acm (could that be the issue here?)

**Comment by slomo** [comment]
This is cause by adhocracy_kit including adhocracy_spd wich both spin up a subscriber, that will than try to set the permissions on the root object, ending up in a conflict.

**Comment by slomo** [comment]
This was caused by having two subscribers (the one from adhocracy_spd and adhocracy_kit) runinng. Which was a result of the kit config including the entire spd config it has been fixed in #1681 and #1689.
